static/app/views/dashboards/controls.tsx

```diff
@@ -55,6 +57,8 @@ function Controls({
   onCancel,
   onAddWidget,
 }: Props) {
+  const [isFavorited, setIsFavorited] = useState(dashboard.isFavorited);
```

Comment:
using a state variable here for better user experience when favouriting, instead of waiting for the api request, making it laggy